src/Executables/Examples/HelloWorld/SingletonHelloWorld.hpp

```diff
@@ -51,8 +52,10 @@ struct HelloWorld {
   using const_global_cache_tag_list = tmpl::list<OptionTags::Name>;
   using chare_type = Parallel::Algorithms::Singleton;
   using metavariables = Metavariables;
-  using action_list = tmpl::list<>;
-  using initial_databox = db::DataBox<tmpl::list<>>;
+  using add_options_to_databox = Parallel::AddNoOptionsToDataBox;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<typename Metavariables::Phase,
+                             Metavariables::Phase::Execute, tmpl::list<>>>;
```

Comment:
I've thought about this quite a bit and there is both a technical issue and I think a conceptual one.
- The technical issue is that we need to retrieve the correct DataBox type, which depends on the phase (ish). This can be worked around fairly reasonably.
- I'm currently somewhat in favor of only allowing any parallel component to be active for phases listed. The reason being is that this means the PDAL alias actually tells any user exactly what phases to be active in. The downside of this is it might make some of the general component code (I'm thinking about observers specifically) depend on the phase, and therefore the executable. This could be worked around easily by taking a `tmpl::value_list` as a template parameter and that being the phases the component is active in.